python/lsst/ts/scriptqueue/ui/request_cmd.py

```diff
@@ -0,0 +1,385 @@
+        If it fails to quit the queue
+        """
+        try:
+            retval = self.model.quit_queue()
+            if retval:
+                print("Bye...")
+            else:
+                print("Could not quit queue.")
+        except Exception as e:
+            print("Could not quit queue.")
+            self.log.exception(e)
+            retval = False
+
+        return retval
+
+    def do_stop(self, args):
+        """Stop list of scripts.
+
+        Parameters
+        ----------
+        args : str
+            Comma separated list of scripts indices.
+        """
+        print(f'Stopping script {args}.')
+        try:
+            sal_indices = [int(i) for i in args.split(",")]
+            self.model.stop_scripts(sal_indices, terminate=False)
+        except Exception as e:
+            self.log.exception(e)
+
+        return 0
+
+    def do_terminate(self, args):
+        """Terminate list of scripts. Similar to stop but will set `terminate` flag to True.
+
+        Parameters
+        ----------
+        args : str
+            Comma separated list of scripts indices.
+        """
+        print(f'Terminating script {args}')
+        try:
+            sal_indices = [int(i) for i in args.split(",")]
+            self.model.stop_scripts(sal_indices, terminate=True)
+        except Exception as e:
+            self.log.exception(e)
+
+        return 0
+
+    def do_show(self, args):
+        """Show current state information about the queue."""
+
+        print("#\n# Showing tasks in the queue.\n#")
+
+        try:
+            queue_state = self.model.get_queue_state()
+
+            print(f'\tQueue state: {queue_state["state"]}')
+            print(f'\tCurrent running: {self.model.parse_info(queue_state["current"])}')
+            print(f'\tCurrent queue size: {queue_state["length"]}')
+            print(f'\tPast queue size: {queue_state["past_length"]}')
+
+            if queue_state['length'] > 0:
+                print(f'\nItems on queue:')
+                for item in queue_state['queue_scripts']:
+                    print(f"{item}: {self.model.parse_info(queue_state['queue_scripts'][item])}")
+            else:
+                print('\nNo items on queue.')
+
+            if queue_state["past_length"] > 0:
+                print(f'\nItems on past queue:')
+                for item in queue_state['past_scripts']:
+                    print(self.model.parse_info(queue_state['past_scripts'][item]))
+            else:
+                print('\nNo items on past queue.')
+
+        except Exception as e:
+            self.log.exception(e)
+
+        return 0
+
+    def do_monitor_script_state(self, salindex):
+        """Monitor a script state until it is done.
+
+        Parameters
+        ----------
+        salindex : int
+            The salindex of the script to monitor.
+        """
+        self.monitor_script(int(salindex), self.model.cmd_timeout)
```

Comment:
I assume this is a command for the same reason as do_show. It would be nice if we could always see the state of the currently running script and report any failures in loading a script (every script that the user added or every script anyone adds; I doubt there will be much difference most of the time when using this UI).